# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/Parser.cpp
# Commit: 73cc5fa0e0bd
# Full Hash: 73cc5fa0e0bdd7a1d6b0f8450b1541c4bc53c333
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 9: Use TaggedParserAtomIndex in RegExpStencil.atom_. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96916
# ==============================================================================

diff -r f505b8fb8f8b -r 73cc5fa0e0bd js/src/frontend/Parser.cpp
--- a/js/src/frontend/Parser.cpp	Sat Nov 14 00:03:07 2020 +0000
+++ b/js/src/frontend/Parser.cpp	Sat Nov 14 00:03:20 2020 +0000
@@ -10301,7 +10301,8 @@
   atom->markUsedByStencil();
 
   RegExpIndex index(this->getCompilationInfo().stencil.regExpData.length());
-  if (!this->getCompilationInfo().stencil.regExpData.emplaceBack(atom, flags)) {
+  if (!this->getCompilationInfo().stencil.regExpData.emplaceBack(
+          atom->toIndex(), flags)) {
     js::ReportOutOfMemory(cx_);
     return nullptr;
   }